### PR TITLE
React Week5 Homework

### PR DESCRIPTION
**Pull request by CharlesKimMaina**

### CODE REVIEW

File: react/week5/githubusersprofile/src/Components/Profile.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const { userName } = useParams();
+
+  //useState is for tracking user's properties and their changes.
+  //useParams returns an object of key/value pairs of URL parameters, in ourcase is "userName"
+  //param is loaded inside profile component, hence its imported in this component.
+  //useEffect is for fetching.
+
+  useEffect(() => {
+    if (user && user.id) return; //if there is user and has user.id, then fetch his/her details from github API.
+
```
Comment (cecastosic) [inline react/week5/githubusersprofile/src/Components/Profile.js:16]:
```suggestion
if (userName) {
```
fetch data if you have userName

File: react/week5/githubusersprofile/src/Components/Profile.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    avatar_url,
+    company,
+    followers,
+    following,
+    public_repos,
+  } = loadedUser; //these are user's object properties as per githubs data dashboard.
+  return (
+    <div className="profile">
+      <div>
+        <img src={avatar_url} alt="" />
```
Comment (cecastosic) [inline react/week5/githubusersprofile/src/Components/Profile.js:38]:
```suggestion
        <img src={avatar_url} alt="avatar picture" />
```
always write alt as description

File: react/week5/githubusersprofile/src/Components/Profile.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import "./Profile.css";
+
+export function Profile({ user }) {
```
Comment (cecastosic) [inline react/week5/githubusersprofile/src/Components/Profile.js:5]:
```suggestion
export function Profile() {
```
Can't see why you need user

File: react/week5/githubusersprofile/src/Components/Profile.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  const [loadedUser, setLoadedUser] = useState(user);
+  const { userName } = useParams();
+
+  //useState is for tracking user's properties and their changes.
+  //useParams returns an object of key/value pairs of URL parameters, in ourcase is "userName"
+  //param is loaded inside profile component, hence its imported in this component.
+  //useEffect is for fetching.
+
+  useEffect(() => {
+    if (user && user.id) return; //if there is user and has user.id, then fetch his/her details from github API.
```
Comment (cecastosic) [inline react/week5/githubusersprofile/src/Components/Profile.js:15]:
```suggestion
    
```

File: react/week5/githubusersprofile/src/Components/Users.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  /*User Context is for figuring is the user has an id. 
+  The first useState does optimisation by tracking the user as a whole then later 
+  i can destructure user's properties.
+  second useState if for tracking errors, invalid user, nameless users, blank search (user.length ==0;)etc 
+  and setError with error message "The user you are searching is not found". */
+
+  useEffect(() => {
+    if (!userId) return;
+
+    fetch(`https://api.github.com/search/users?q=${userId}`)
```
Comment (cecastosic) [inline react/week5/githubusersprofile/src/Components/Users.js:18]:
```suggestion
    if (userId) {
    fetch(`https://api.github.com/search/users?q=${userId}`)
```

File: react/week5/githubusersprofile/src/Components/Users.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      .then((res) => res.json())
+      .then((res) => {
+        setUsers(res.items);
+      })
+      .catch((error) => setError(error.message));
+  }, [userId]);
+
+  if (!userId) return "The user you are searching is not found";
```
Comment (cecastosic) [inline react/week5/githubusersprofile/src/Components/Users.js:26]:
```suggestion
  if (!userId) return "Type a name in the search input";
```

File: react/week5/githubusersprofile/src/Components/Profile.css [hunk 7]
```diff
@@ -0,0 +1,18 @@
+div.profile {
```
Comment (cecastosic) [inline react/week5/githubusersprofile/src/Components/Profile.css:1]: this file shouldn't be part of components... put Profile.js and Profile.css in a folder named Profile in Components folder

**Review by cecastosic — COMMENTED** [review]
Nice 💪 I would work more on styling 🙂 Some suggestions from my side 👇

File: react/week5/githubusersprofile/src/Components/Profile.css (hunk 7 above)
Comment (CharlesKimMaina) [reply react/week5/githubusersprofile/src/Components/Profile.css:1]: yah, i get it,.Thanks

**Pull request merged**